### DISCUSSION TITLE
Can't deploy on Vercel with Next.js

### DISCUSSION THREAD

**Discussion by FelipeBattistottiAvant** [vercel/next.js#34741] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Enterprise
    Binaries:
      Node: 16.0.0
      npm: 7.10.0
      Yarn: 1.22.5
      pnpm: N/A
    Relevant packages:
      next: 12.1.0
      react: 18.0.0-rc.0-next-fe905f152-20220107    
      react-dom: 18.0.0-rc.0-next-fe905f152-20220107

### What browser are you using? (if relevant)

Chrome 98.0.4758.102

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

I can't deploy on Vercel with Next.js,
the following message appears:

Build Failed
Your deployment's build step did not complete within the maximum of 45 minutes (task 4d4091ed3b274d83acdf436057a9e11f)

Logs from build:
Cloning github.com/AvantSoftwares/plenusweb-frontend (Branch: master, Commit: 4c639aa)
Cloning completed: 1.984s
Analyzing source code...
Installing build runtime...
Build runtime installed: 2.495s
Looking up build cache...
Build cache downloaded [102.91 MB]: 7991.657ms
Installing dependencies...
yarn install v1.22.17
info No lockfile found.
[1/4] Resolving packages...
warning @material-ui/core@4.12.3: You can now upgrade to @mui/material. See the guide: https://mui.com/guides/migration-v4/
warning @material-ui/core > @material-ui/styles@4.11.4: You can now upgrade to @mui/styles. See the guide: https://mui.com/guides/migration-v4/
warning @material-ui/core > @material-ui/system@4.12.1: You can now upgrade to @mui/system. See the guide: https://mui.com/guides/migration-v4/
warning @material-ui/lab@4.0.0-alpha.57: You can now upgrade to @mui/lab. See the guide: https://mui.com/guides/migration-v4/
warning @material-ui/styles@4.11.4: You can now upgrade to @mui/styles. See the guide: https://mui.com/guides/migration-v4/
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning " > @material-ui/core@4.12.3" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @material-ui/core@4.12.3" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @material-ui/styles@4.11.4" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @material-ui/styles@4.11.4" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/system@4.12.1" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/system@4.12.1" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/utils@4.11.2" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/utils@4.11.2" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @material-ui/lab@4.0.0-alpha.57" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @material-ui/lab@4.0.0-alpha.57" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @material-ui/pickers@3.3.10" has unmet peer dependency "@date-io/core@^1.3.6".
warning " > @material-ui/pickers@3.3.10" has unmet peer dependency "prop-types@^15.6.0".
warning " > @material-ui/pickers@3.3.10" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @material-ui/pickers@3.3.10" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @mui/icons-material@5.4.2" has incorrect peer dependency "react@^17.0.0".
warning " > @mui/material@5.4.3" has incorrect peer dependency "react@^17.0.0".
warning " > @mui/material@5.4.3" has incorrect peer dependency "react-dom@^17.0.0".
warning "@mui/material > @mui/base@5.0.0-alpha.69" has incorrect peer dependency "react@^17.0.0".
warning "@mui/material > @mui/base@5.0.0-alpha.69" has incorrect peer dependency "react-dom@^17.0.0".
warning "@mui/material > @mui/system@5.4.3" has incorrect peer dependency "react@^17.0.0".
warning "@mui/material > @mui/system > @mui/styled-engine@5.4.2" has incorrect peer dependency "react@^17.0.0".
warning "@mui/material > @mui/utils@5.4.2" has incorrect peer dependency "react@^17.0.0".
warning "@mui/material > @mui/system > @mui/private-theming@5.4.2" has incorrect peer dependency "react@^17.0.0".
warning " > @tailwindcss/custom-forms@0.2.1" has incorrect peer dependency "tailwindcss@^1.0".
warning " > babel-loader@8.2.3" has unmet peer dependency "webpack@>=2".
warning "devextreme > @devextreme/runtime@2.3.14" has unmet peer dependency "@angular/core@^11.0.4".
warning "devextreme > @devextreme/runtime@2.3.14" has unmet peer dependency "vue@^2.6.14".
warning " > devextreme-react@21.2.5" has incorrect peer dependency "react@^16.2.0 || ^17.0.0".
warning " > devextreme-react@21.2.5" has incorrect peer dependency "react-dom@^16.2.0 || ^17.0.0".
warning "next > use-subscription@1.5.1" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > react-alert@7.0.3" has incorrect peer dependency "react@^16.8.1 || ^17".
warning " > react-alert@7.0.3" has incorrect peer dependency "react-dom@^16.8.1 || ^17".
warning " > react-data-grid@7.0.0-canary.33" has incorrect peer dependency "react@^16.14 || ^17.0".
warning " > react-data-grid@7.0.0-canary.33" has incorrect peer dependency "react-dom@^16.14 || ^17.0".
warning " > react-swipeable-views@0.13.9" has incorrect peer dependency "react@^15.3.0 || ^16.0.0".
warning "react-swipeable-views > react-swipeable-views-utils > react-event-listener@0.6.6" has incorrect peer dependency "react@^16.3.0".
warning " > styled-components@5.3.3" has unmet peer dependency "react-is@>= 16.8.0".
warning " > swr@0.5.7" h

![image](https://user-images.githubusercontent.com/71387673/155387470-62bd6f7d-e76b-41e6-bb1d-69b3e0828626.png)

![image](https://user-images.githubusercontent.com/71387673/155387527-0200a727-57c1-4e3c-8574-a773882b7e31.png)

![image](https://user-images.githubusercontent.com/71387673/155387610-28e23393-c0db-4ce0-90e7-e66b0cfca984.png)

### Expected Behavior

Deploy on Vercel

### To Reproduce

Just deploy Next.js app on Vercel

**Comment by balazsorban44** [comment]
There have been no known incidents, so could you please attach a reproduction so we could look more into this? The repository from the logs seems to be private so I cannot have a look.

**Reply by FelipeBattistottiAvant** [reply]
@balazsorban44 
Balázs,
I forgot mention, this problem occurs only integration by GitHub.
If deploy exceed 45 minutes, then it breaks and show the message:

"Your deployment's build step did not complete within the maximum of 45 minutes (task 8df143bbee664ac29ba9de2088be7117)"

Why this occurs only on integration by GitHub?

**Reply by balazsorban44** [reply]
It's hard to tell, could you please share the code that triggers this? Is this consistently happening or only happening sporadically/one time?

**Reply by FelipeBattistottiAvant** [reply]
@balazsorban44 
Is constantly happening.
I believe that solution should come from Vercel.

**Reply by balazsorban44** [reply]
We would definitely like to investigate, but we need to see your code. If your project is private, could you invite me to it to have a look? :pray:

**Reply by FelipeBattistottiAvant** [reply]
@balazsorban44 
I solved it myself.

Thanks Balázs

**Reply by deadinsiDEV** [reply]
how to fixed it i get same error

**Comment by cronin4392** [comment]
I am running into this as well on a project. Issue started occurring yesterday and now any build triggered through webhooks from Github timeout during the installing dependencies phase. I've confirmed that the build works locally with both `next build` and `vercel build --prod`. I also confirmed that triggering a build with `vercel deploy` works successfully. I've tried doing a redeploy in the Vercel portal both with and without build cache but the same behavior occurs.

The next thing I tried was bumping my version of Next.js to the latest but still getting the timeout during installing dependencies. Only other solution I've read in some threads is to create a whole new Vercel project, but I'd really prefer to not have to set that all up again. I've run into similar issues in the past which magically seemed to resolve themselves.

**Comment by ChrisLeeRios** [comment]
How did you fix this? im running into the same exact issue and its really wracking my brain

**Reply by cronin4392** [reply]
The issue eventually just went away for us I think. I've run into these types of issues sporadically and never found the exact culprit.

**Comment by lgaieta** [comment]
Hi, I had the same problem and I found the solution here: https://nextjs.org/docs/advanced-features/compiler#modularize-imports
I realized MUI imports were the problem. Specially the default imports with "@mui/material", that I used in the whole app.
When I applied modularized imports to MUI, the compiling time reduced a lot. I saw there is also a babel plugin that works the same. Hope you find it helpful.
